new feature dynamic IP's list

Omer-Sadeh opened the pull request:
replaced the dropdown menu of IPs with a pop up window that opens when pressing the button, which inside we have a dynamic list, and an option to add new IPs and delete existing ones.
Original dropdown menu:
![old dropdown menu](https://user-images.githubusercontent.com/88449326/133074286-88cb05cc-600d-47cc-b279-6f84ce348d9e.PNG)
![old ui](https://user-images.githubusercontent.com/88449326/133080539-78571cc8-7c6e-48e6-9237-f423f46776b2.PNG)
New Dynamic List:
![new dynamic IP list](https://user-images.githubusercontent.com/88449326/133075663-d330eae1-711a-4337-a740-1c819854b5e1.PNG)
the button:
![new ui 1](https://user-images.githubusercontent.com/88449326/133080594-b3e2ee00-ffaa-48cf-ac99-c73f8d8392d4.PNG)
the new pop up window:
![new ui 2](https://user-images.githubusercontent.com/88449326/133080633-cf83ddc4-1908-4223-bdbb-4227d3db47f5.PNG)
and the delete button, after pressing it:
![new ui 3](https://user-images.githubusercontent.com/88449326/133080672-87369e50-1539-47f7-b0c3-618179f4d656.PNG)
In order to do that, I modified the states of App, changing the name of 2 of them to fit better with current code, and adding 2 (1 toggle of the pop up window, Modal, and 1 which is an parallel array to the IP's array, for the Delete button I added):
![states edit](https://user-images.githubusercontent.com/88449326/133074279-f0a4a943-36be-4b56-9676-d6af78f20985.PNG)
Also, I made new functions, for the toggles and switching:
![new func 1](https://user-images.githubusercontent.com/88449326/133074283-dd63eb4e-485d-4bb1-be37-9a714dca57ef.PNG)
and for the buttons themselves and the deletion operation:
![new func 2](https://user-images.githubusercontent.com/88449326/133075722-f38f3b5d-b154-45c6-aa6c-d7018b4e54ab.PNG)

Omer-Sadeh committed:
Update App.tsx

Omer-Sadeh committed:
Update App.tsx

Omer-Sadeh committed:
Update App.tsx

Omer-Sadeh committed:
Update App.tsx

lomudi commented:
@Omer-Sadeh 
1. please notice that the CI deploy tests where failed, please fix it.
2. about the PR description - it is great that you are providing all the details, in general code changes can be easily seen under the "file changed" tab, so no need to put screenshots of code changes... in the other hand, screenshots of the UI changes are very valuable for the reviewer - please add those to the PR description. 

Thx! I have asked @EtayTod to review the PR.

Omer-Sadeh committed:
Update App.tsx

Omer-Sadeh committed:
Revert "Update App.tsx"

This reverts commit 00a1d45c871c6c6e008a04a3667040345d2edc24.

Omer-Sadeh committed:
Revert "Update App.tsx"

This reverts commit 1da4750b54e0abd3f6aa9f88468c4ce355e62d51.

Omer-Sadeh committed:
Update App.tsx

Omer-Sadeh committed:
removed unnecessary imports.

Omer-Sadeh committed:
Update RobotObject.tsx

Omer-Sadeh commented:
fixed the CI deploy issues, it required removing un-used imports and a small fix in RobotObject.tsx.
also, added more screenshots to description.

Todress commented on src/App.tsx:
```diff
@@ -2,21 +2,21 @@ import React from 'react';
 import './App.scss';
 import { HttpClient } from '@butter-robotics/mas-javascript-api';
 import { RobotObject } from './components/RobotObject';
-import { useState } from 'react';
```
why removing this import?

Omer-Sadeh replied:
The PR deploy test failed when I had warnings in the code, such as "defined but never used", so i have to delete this never used line.

Todress commented on src/App.tsx:
```diff
@@ -60,9 +56,65 @@ export class App extends React.PureComponent<{}, AppState> {
+			StateArray[index] = false;
+			return(
+				<>
+				<Button variant="secondary" disabled>Are you sure you want to delete {ip}?</Button>
```
look at the css of other buttons, we should follow one line of design here

Omer-Sadeh replied:
would love some more explainations on this.

Todress commented on src/components/RobotObject.tsx:
```diff
@@ -9,7 +9,7 @@ export function RobotObject({ butterClient, onRemove }: { butterClient: HttpClie
 
   useEffect(() => {
     loadAnimations();
-  }, []);
```
why removing []?

Omer-Sadeh replied:
The PR deploy test failed when I had warnings in the code, and this line had a warning for unnecessary [].

Todress commented on src/components/ScenariosButtons.tsx:
```diff
@@ -1,16 +1,13 @@
 kip2.timeout = 240;
 
-let interval1: NodeJS.Timeout;
-let interval2: NodeJS.Timeout;
```
aren't we using those intervals? did you perform a live test on few robots to verify all functioning well?

Omer-Sadeh replied:
The PR deploy test failed when I had warnings in the code, such as "defined but never used", so i have to delete this never used line. 
I didn't check it in the lab, but we can keep this lines, as I said, only removed them to get rid of the warnings to pass the deploy tests.

Todress commented on src/index.tsx:
```diff
@@ -3,7 +3,6 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
-import { BrowserRouter, Route, Switch } from "react-router-dom"
```
same - aren't we using this?

Omer-Sadeh replied:
The PR deploy test failed when I had warnings in the code, such as "defined but never used", so i have to delete this never used line.

Todress commented on src/App.tsx:
```diff
@@ -25,10 +25,12 @@ export class App extends React.PureComponent<{}, AppState> {
+		showInst: false,
+		showNewIP: false,
 		labCurrentIPs: ['192.168.57.30', '192.168.57.32', '192.168.57.34', '192.168.56.188', '192.168.56.193', '192.168.56.206'],
+		IPdeleteState: [false, false, false, false, false, false],
```
what happens to this list after you delete and then add new ip?

Todress replied:
make sure that you don't mix states with different ip's after deleting and adding some

Omer-Sadeh replied:
you can see in the code, where you add and remove IPs, it updates the new array as well, and I tested it and it works.
It checks the index of the IP in the IPs array, and when removing it, it removes the corresponding index in the states array.

Todress reviewed (changes requested):
please note my comments, if something is not clear - reply here to the comments.
beside the code review, did you perform an actual test with the robots?
beside that - looking great! 👍

Omer-Sadeh committed:
ip input validation & fix delete button css

Omer-Sadeh committed:
expanded IP validation

Omer-Sadeh committed:
Update App.tsx

Omer-Sadeh committed:
fixed warnings

Omer-Sadeh committed:
Revert & fix Line

Reverting all the lines that was deleted, and fixing syntax of a line.

Omer-Sadeh commented:
Added back all the lines I deleted that caused warnings, and changed the syntax in a line in RobotObject component, to be according to the same line in Develop, as seen in the picture:
![Capture](https://user-images.githubusercontent.com/88449326/134814933-35cdb42e-98f0-42f3-9235-ba847cf1bc78.PNG)

Todress reviewed (approved):
Approved
